tests/queries/tests.py

```diff
@@ -1173,6 +1173,23 @@ def test_common_mixed_case_foreign_keys(self):
         )
         self.assertTrue(qs.first())
 
+    def test_excluded_intermediary_m2m_table_joined(self):
+        # Three JOINs on "note" column.
+        self.assertSequenceEqual(
+            Note.objects.filter(~Q(tag__annotation__name=F('note'))),
+            [self.n1, self.n2, self.n3],
+        )
+        self.assertSequenceEqual(
+            Note.objects.filter(
+                tag__annotation__name='a1',
+            ).filter(
+                tag__annotation__name='a1',
```

Comment:
Was this meant to filter twice on `='a1'`?

Reply:
You're right it is not necessary to reproduce this issue.